### DISCUSSION TITLE
How can I get client Ip Address?

### DISCUSSION THREAD

**Discussion by NepPure** [justcoding121/Titanium-Web-Proxy#928] [Q&A] [answered]
I didn't find such property in Request.

Thanks!

**Comment by justcoding121** [comment, marked as answer]
It's on the session, not the request:

```csharp
proxyServer.BeforeRequest += async (sender, e) =>
{
    var clientIp = e.ClientRemoteEndPoint.Address;
    Console.WriteLine($"Client IP: {clientIp}");
};
```

`e.ClientRemoteEndPoint` is the TCP peer that connected to the proxy. (There's also `e.ClientLocalEndPoint` for the local side of that socket.)

Closing this one out.
